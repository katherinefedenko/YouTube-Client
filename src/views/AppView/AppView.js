export default class AppView {
  constructor(titles) {
    this.titles = titles;
    this.watchURL = 'https://www.youtube.com/watch?v=';
  }

  render() {
    const {
      channelTitle, videoTitle, videoImage, videoDate, videoDescription, videoId,
    } = this.titles;
    const clipWrapper = document.createElement('div');
    clipWrapper.className = 'clips__wrapper';
    document.body.appendChild(clipWrapper);

    const icons = document.createElement('link');
    icons.innerHTML = '<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.2/css/all.css" integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay" crossorigin="anonymous">';
    document.head.appendChild(icons);
    // eslint-disable-next-line no-plusplus
    for (let i = 0; i < 15; i++) {
      const clip = document.createElement('div');
      clip.className = 'clip';
      clipWrapper.appendChild(clip);

      const vTitle = document.createElement('div');
      vTitle.innerHTML = `<a href="${this.watchURL}${videoId[i]}">${videoTitle[i]}</a>`;

      const chTitle = document.createElement('div');
      chTitle.className = 'info__wrapper';
      chTitle.innerHTML = `<i class="icon fas fa-user"></i><div>${channelTitle[i]}</div>`;

      const vImage = document.createElement('div');
      vImage.innerHTML = `<img src="${videoImage[i]}">`;

      const vDate = document.createElement('div');
      vDate.className = 'info__wrapper';
      vDate.innerHTML = `<i class=" icon far fa-calendar-alt"></i><div>${videoDate[i].slice(0, 10)}</div>`;

      const vDescription = document.createElement('div');
      vDescription.innerHTML = `<div>${videoDescription[i]}</div>`;

      clip.appendChild(vImage);
      clip.appendChild(vTitle);
      clip.appendChild(chTitle);
      clip.appendChild(vDate);
      clip.appendChild(vDescription);
    }
  }
}
