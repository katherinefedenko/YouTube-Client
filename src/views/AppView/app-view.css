@import url('https://fonts.googleapis.com/css?family=Lora');
@import url('https://use.fontawesome.com/releases/v5.8.2/css/all.css" integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay" crossorigin="anonymous"');
body {
  background-color: #777777;
  font-family: 'Lora', sans-serif;
}
