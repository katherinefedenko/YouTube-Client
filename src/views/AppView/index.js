import AppView from './AppView';

import './app-view.css';
import '../ClipsView/clipView.css';
import '../SearchBoxView/searchBox.css';

export default AppView;
