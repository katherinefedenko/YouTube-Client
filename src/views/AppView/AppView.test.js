import AppView from './AppView';

describe('AppView.prototype.render', () => {
  it('Should be an instance of a function', () => {
    expect(AppView.prototype.render).toBeInstanceOf(Function);
  });

  it('Should be render correctly', () => {
    const context = {
      titles: [
        'Video about JS',
      ],
    };
    AppView.prototype.render.call(context);
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
