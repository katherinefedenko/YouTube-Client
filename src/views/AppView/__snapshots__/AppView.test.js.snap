// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.prototype.render Should be render correctly 1`] = `"<ul><li>Video about JS</li></ul>"`;
